### PR TITLE
Refactor github_hook module

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
* PEP8 fixes
* Refactoring for fetch_url call
* Removed un-used code

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/source_control/github_hooks.py
test/sanity/pep8/legacy-files.txt

##### ANSIBLE VERSION
```
2.4 devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
ready_for_review

File: lib/ansible/modules/source_control/github_hooks.py [hunk 1]
```diff
@@ -146,37 +146,30 @@ def _create(module, hookurl, oauthkey, repo, user, content_type):
...
 def main():
     module = AnsibleModule(
         argument_spec=dict(
-            action=dict(required=True, choices=['list','clean504','cleanall','create']),
+            action=dict(required=True, choices=['list', 'clean504', 'cleanall', 'create']),
             hookurl=dict(required=False),
             oauthkey=dict(required=True, no_log=True),
             repo=dict(required=True),
             user=dict(required=True),
-            validate_certs=dict(default='yes', type='bool'),
```
Comment (resmo) [inline lib/ansible/modules/source_control/github_hooks.py]: What is the reason of change? it would break the backwards compatibility.

**Comment by resmo** [comment]
needs_info

**Comment by Akasurde** [comment]
@resmo I removed code block by mistake. See update.

**Comment by resmo** [comment]
Thanks @Akasurde, code LGTM.

shipit

**Review by resmo — APPROVED** [review]

**Comment by Akasurde** [comment]
ready_for_review

**Comment by Akasurde** [comment]
@resmo Thanks for review.

**Pull request merged**